index.js

```diff
@@ -33,6 +34,15 @@ function logid (pkg, stage) {
   return pkg._id + '~' + stage + ':'
 }
 
+function hookExists (dir, stage) {
+  const hook = path.join(dir, '.hooks', stage)
+  if (_hookExistsCache[hook] === undefined) {
+    _hookExistsCache[hook] = fs.existsSync(hook)
```

Comment:
I'd feel much more comfortable if this were an async operation. Also, `fs.exists` is deprecated in Node, so you'll have to use `fs.stat` with a catcher for `err.code === 'ENOENT'`.

Reply:
I think sync is fine here considering the cache may only get 3 or 4 values in it (opts.dir is constant, so the size of the cache is like 3 or 4), but happy to go async. I’ll do the change.